fennel/client_tests/test_dataset.py

```diff
@@ -1206,6 +1285,66 @@ def test_basic_aggregate(self, client):
         assert df["distinct_users"].tolist() == [[18231], [18231]]
 
 
+class TestBasicAggregateAlong(unittest.TestCase):
+    @pytest.mark.integration
+    @mock
+    def test_basic_aggregate_along(self, client):
+        # # Sync the dataset
+        client.commit(
+            message="msg",
+            datasets=[MovieRatingCalculatedAlong, RatingActivityAlong],
+        )
+        now = datetime.utcnow()
+        one_hour_ago = now - timedelta(hours=1)
+        two_hours_ago = now - timedelta(hours=2)
+        three_hours_ago = now - timedelta(hours=3)
+        four_hours_ago = now - timedelta(hours=4)
+        five_hours_ago = now - timedelta(hours=5)
+
+        data = [
+            [18231, 2, "Jumanji", five_hours_ago, five_hours_ago],
+            [18231, 3, "Jumanji", five_hours_ago, four_hours_ago],
+            [18231, 2, "Jumanji", five_hours_ago, three_hours_ago],
+            [18231, 5, "Jumanji", five_hours_ago, five_hours_ago],
+            [18231, 4, "Titanic", five_hours_ago, three_hours_ago],
+            [18231, 3, "Titanic", five_hours_ago, two_hours_ago],
+            [18231, 5, "Titanic", five_hours_ago, one_hour_ago],
+            [18231, 5, "Titanic", five_hours_ago, now - timedelta(minutes=1)],
+            [18231, 3, "Titanic", five_hours_ago, two_hours_ago],
+        ]
+        columns = ["userid", "rating", "movie", "t", "rating_time"]
+        df = pd.DataFrame(data, columns=columns)
+        response = client.log("fennel_webhook", "RatingActivity", df)
+        assert response.status_code == requests.codes.OK
+
+        client.sleep()
+
+        # Do some lookups to verify pipeline_aggregate is working as expected
+        ts = pd.Series([now, now])
```

Comment:
Could you please test this on some other timestamps too such that more complex interaction of along and timestamp are tested?

Reply:
Will add more test query back in time but still following the hacky along like server side right now